ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -614,23 +617,25 @@ private CountDownLatch submitPut()
       throws Exception {
     final CountDownLatch doneLatch = new CountDownLatch(requestAndResultsList.size());
     // This check is here for certain tests (like testConcurrentPuts) that require using the same router.
-    if (router == null || !router.isOpen()) {
+    if (instantiateNewRouterForPuts) {
```

Comment:
Ok, as long as we make sure we make sure we use the same instance all the time in the end to end to tests that @cgtz is working on, this should be ok.